src/client/stdlib/duckdb.js

```diff
@@ -178,9 +189,38 @@ export class DuckDBClient {
   }
 }
 
-Object.defineProperty(DuckDBClient.prototype, "dialect", {
-  value: "duckdb"
-});
+Object.defineProperty(DuckDBClient.prototype, "dialect", {value: "duckdb"});
+
+async function registerExtensions(db, {load}) {
+  const connection = await db.connect();
+  try {
+    // Preview adds a DuckDBClientReport utility to the console. We don’t add it
+    // in the public build so as not to pollute the window.
+    if (manifest.log) {
+      window.DuckDBClientReport = async () => {
+        const connection = await db.connect();
+        try {
+          const refs = new Map(extensions);
+          const ext = await connection.query(
+            "SELECT extension_name, description FROM duckdb_extensions() WHERE loaded;"
+          );
+          console.table(Array.from(ext, (e) => ({...e, ...refs.get(e.extension_name)})));
+        } finally {
+          await connection.close();
+        }
+      };
+    }
```

Comment:
I think we should remove this as I noted above. I think we can just document “run this SQL if you want to know what DuckDB extensions are loaded” and not make a helper method for it.

Principles:

- Don’t pollute global scope
- Functions should be bound to appropriate scope (here the DuckDBClient/`db` instance)
- Minimize differences in logic between production and development
- Favor minimal API surface area so it’s easier for us to keep things light
- Favor methods that return data, not that have a side-effect of console.log/console.table

If we also want to expose the manifest/registered extensions and not just what DuckDB sees, then it should be a method on DuckDBClient. Or if want to expose the manifest, it should be a static property on DuckDBClient. But better to minimize what we expose so we have more flexibility to change it in the future, and there’s less for us to document, and less for readers to read.

Reply:
It was very useful to me while debugging, but there's no point in keeping this. The network tab is enough to know which extensions are loaded.